src/types/core.d.ts

```diff
@@ -233,6 +233,7 @@ export interface TStyledSheet<
   utils: C;
 
   theme: (theme: Partial<{ [TO in keyof B]: Partial<B[TO]> }>) => ThemeRule;
+  useTheme: () => Partial<{ [TO in keyof B]: Partial<B[TO]> }>;
```

Comment:
It seems that the partial type used as argument in the `theme` function is used to override some (nested) fields of the initial theme and we can be more accurate in the return type of `useTheme` by just returning the type of the theme itself. This would also get rid of the optional chaining (`theme.colors?.blue100` becomes `theme.colors.blue100`).

```suggestion
  useTheme: () => B;
```

Note that this is still not the fully correct type since resolved tokens might not end up being strings anymore, e.g.:
```ts
// config
{ space: { s: 8, alsoS: "$s" } } // -> { space: { s: number; alsoS: string }

// useTheme output
{ space: { s: 8, alsoS: 8 } } // should be { space: { s: number; alsoS: number }
```

I'm wondering if we can resolve the type of the tokens with some conditional type magic and produce the actual type of the theme.

Reply:
@finkef raises a valid point. I think it could be a reason to do something like this instead
https://github.com/Temzasse/stitches-native/issues/2#issuecomment-892666039
So we let the internals do its magic to resolve tokens, aliases, etc; instead of accessing the theme raw values directly

Reply:
Personally, I think it would be much cleaner to do it this way with `useTheme()` and directly consume the resolved values without the need to go through `processStyles` (also with respect to limiting the calls to `processStyles` in every render). Only challenge is to get the return type right. I'm not sure if the core stitches lib has any helpers or types to already do this, I can take a look into this this weekend.

Reply:
Yeah I think having to work with `$token` creates an unnecessary indirection that can be avoided by using the raw values from `useTheme`.

Related to the TS type issue: Stitches has [recently](https://github.com/modulz/stitches/pull/659) rewritten their TS types and we should at some point try to bring over the new types. It was quite a hurdle to bring over the types initially so I'm a bit hesitant to spend time on it now (altho the new types look much more cleanly structured).

Reply:
Yes. I can't wait to see those TS types released.
For now we could keep `useTheme: () => B` and use the raw values, trying to avoid usages of the aliases.